broker_test.go

```diff
@@ -383,3 +379,57 @@ func TestSuccessThresholdSinks(t *testing.T) {
 		t.Fatalf("expected successThresholdSinks error")
 	}
 }
+
+// TestRemovePipelineAndNodes exercises the behavior that removes a pipeline and
+// any nodes associated with that pipeline, if they are not referenced by other pipelines.
+func TestRemovePipelineAndNodes(t *testing.T) {
```

Comment:
do we want tests for failure modes of: 
* no graph for event type
* no nodes in a pipeline
* node is not registered